README.md

```diff
@@ -1,50 +1,157 @@
-ember-simple-auth-oidc
-==============================================================================
+# ember-simple-auth-oidc
 
-[Short description of the addon.]
+A [Ember Simple Auth](http://ember-simple-auth.com) addon which implements the
+OpenID Connect [Authorization Code Flow](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth).
 
-Installation
-------------------------------------------------------------------------------
+## Installation
 
 ```
 ember install ember-simple-auth-oidc
 ```
 
+## Usage
 
-Usage
-------------------------------------------------------------------------------
+To use the oidc authorization code flow add at least the following mixins to
+their respective routes:
 
-[Longer description of how to use the addon in apps.]
+The `oidc-application-route-mixin` replaces the Ember Simple Auth `application-route-mixin`.
 
+```
+// app/routes/application.js
+import Route from "@ember/routing/route";
+import OIDCApplicationRouteMixin from "ember-simple-auth-oidc/mixins/oidc-application-route-mixin";
+
+export default Route.extend(OIDCApplicationRouteMixin, {});
+```
+
+The `oidc-authentication-route-mixin` must cover the login / authentication route
+(for example the Ember Simple Auth default `/login`).
+
+```
+// app/routes/login.js
+import Route from "@ember/routing/route";
+import OIDCAuthenticationRouteMixin from "ember-simple-auth-oidc/mixins/oidc-authentication-route-mixin";
+
+export default Route.extend(OIDCAuthenticationRouteMixin, {});
+```
+
+The `oidc-end-session-route-mixin` must cover the logout / end session route
+(for example `/logout`).
+
+```
+// app/routes/logout.js
+import Route from "@ember/routing/route";
+import OIDCEndSessionRouteMixin from "ember-simple-auth-oidc/mixins/oidc-end-session-route-mixin";
+
+export default Route.extend(OIDCEndSessionRouteMixin, {});
+```
+
+To include authorization info in all Ember Data requests add the `oidc-adapter-mixin`
+into the application adapter.
+
+```
+// app/adapters/application.js
+import DS from "ember-data";
+import OIDCAdapterMixin from "ember-simple-auth-oidc/mixins/oidc-adapter-mixin";
+
+export default DS.JSONAPIAdapter.extend(OIDCAdapterMixin, {});
+```
+
+## Configuration
+
+The addon can be configured in the ember config with the key `ember-simple-auth-oidc`.
+
+At least the following config options need to be added to the ember config:
+
+```
+module.exports = function(environment) {
+  let ENV = {
+    modulePrefix: "oidc",
+    environment,
+    rootURL: "/",
+    "ember-simple-auth-oidc": {
+      host: "http://authorization.server/openid",
+      clientId: "test",
+      authEndpoint: "/authorize",
+      tokenEndpoint: "/token",
+      userinfoEndpoint: "/userinfo",
+    }
+  }
+  return ENV;
+}
```

Comment:
I'd focus on only the relevant section, maybe like this:
```js
module.exports = function(environment) {
  let ENV = {
    // ...
    "ember-simple-auth-oidc": {
      host: "http://authorization.server/openid",
      clientId: "test",
      authEndpoint: "/authorize",
      tokenEndpoint: "/token",
      userinfoEndpoint: "/userinfo",
    }
    // ...
  }
}
```

Reply:
Agree, also changed this one as suggested.